lib/Canto/Curs/Utils.pm

```diff
@@ -1327,11 +1327,11 @@ sub make_allele_display_name
 
   $description ||= $type || 'unknown';
 
-  if ($name =~ /$description/) {
+  if ($name =~ /[^a-z\d]$description$/) {
```

Comment:
I'm not sure if this regular expression will match what was intended. It sounds like you wanted to match when the description "is at the end of the allele name and preceded by non letter or number", but this doesn't match numbers because the character class negation (`^`) applies to the _whole_ class. This is what the regex currently means:

![image](https://user-images.githubusercontent.com/37659591/111608921-a91b8600-87d1-11eb-8ff7-81592e41eee8.png)

Also, as far as I can tell, the expression is case sensitive, so it will still match descriptions preceded by an uppercase letter. I think this expression will do what you wanted:

```suggestion
  if ($name =~ /([^A-Za-z]|\d)$description$/) {
```

However, I'm not sure if regular expressions are the right solution here: using the example fan1-D651A(aaD651A), if I assume 'aaD651A' is the description (i.e. 'aa' is part of the description), then presumably this will never match because 'aaD651A' is not a suffix of 'fan1-D651A' – although without the 'aa', it would be. Maybe the match is supposed to be the other way round, but 'fan1-D651A' isn't a prefix of 'aaD651A' either, so I can't see how this will work. If the 'aa' prefix will be a problem, maybe there's a Perl function that can do suffix matching instead?

Reply:
> maybe there's a Perl function that can do suffix matching instead?

Actually I don't even think this would work because as I just said, "aaD651A is not a suffix". Maybe the simplest fix is just to remove 'aa' or 'nt' from the start of the description and then try to match the new string.